gax-java/gax-grpc/src/main/java/com/google/api/gax/grpc/GrpcCallContext.java

```diff
@@ -152,7 +152,16 @@ private GrpcCallContext(
     this.options = Preconditions.checkNotNull(options);
     this.retrySettings = retrySettings;
     this.retryableCodes = retryableCodes == null ? null : ImmutableSet.copyOf(retryableCodes);
-    this.endpointContext = endpointContext;
+    // Attempt to create an empty, non-functioning EndpointContext by default (if set to null).
+    // This will be configured and updated by the client with user configurations. Attempts to
+    // use the default endpointContext (not updated with configurations) will result in incorrect
+    // data back.
+    try {
+      this.endpointContext =
+          endpointContext == null ? EndpointContext.newBuilder().build() : endpointContext;
+    } catch (IOException ex) {
+      throw new RuntimeException(ex);
+    }
```

Comment:
This was added because Java-Logging creates the ClientContext and creates a default CallContext inside the client: https://github.com/googleapis/java-logging/blob/447c82cafb77379a2dffee7155c58a098fa58e83/google-cloud-logging/src/main/java/com/google/cloud/logging/spi/v2/GrpcLoggingRpc.java#L122

Reply:
>  if a handwritten library creates GrpcCallContext directly, then any gRPC calls would fail with null pointer exception

It would if we don't create an empty EndpointContext. It'll NPE when it does something like `ClientContext.getDefaultCallContext().getEndpointContext()...`